### PR TITLE
Opportunity model

### LINKED ISSUE #31: Create Opportunity Model

### LINKED ISSUE #32: Create Opportunity serializers/views/urls

### PR DESCRIPTION
**Pull request by cadecairos**
1. implement Opportunity model
2. implement model serializer for Opportunity
3. implement views for Opportunities
4. add URLs for accessing Opportunities.
5. Genericize the sortable functinality so that Person and Opportunity can share it
6. Add Opportunity to the admin site

Rebase, then land, after #53 

fixes #31
fixes #32

### CODE REVIEW

File: app/networkapi/templates/shared/adminsortable_objects_custom.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <input name="pk" type="hidden" value="{{ object.pk }}" />
     <div id="sortable-object">
         <a href="{% url opts|admin_urlname:'do_sorting' object.model_type_id %}" class="admin_sorting_url">
-            <img class="person-image" src="{{ object.image.url }}">
+            <img class="instance-image" src="{{ object.image.url }}">
```
Comment (gideonthomas) [inline app/networkapi/templates/shared/adminsortable_objects_custom.html]: I actually think we shouldn't genericize this file because it relies on properties that may or may not be part of a model (for e.g. `image` and also soon to be added - `featured`). So, I think it makes sense to duplicate this file for each app.

File: app/networkapi/static/people/js/admin.js [hunk 2]
```diff
@@ -1,11 +0,0 @@
-function run($) {
```
Comment (gideonthomas) [inline app/networkapi/static/people/js/admin.js]: why this change?

File: app/networkapi/people/templatetags/people_adminsortable_tags_custom.py [hunk 3]
```diff
@@ -46,6 +46,6 @@ def render_object_rep(
 ):
     context.update({'object': obj, 'forloop': forloop})
     tmpl = template.loader.get_template(
-        'people/adminsortable_objects_custom.html'
+        'shared/adminsortable_objects_custom.html'
```
Comment (gideonthomas) [inline app/networkapi/people/templatetags/people_adminsortable_tags_custom.py]: let's undo this based on my suggestion about duplicating templates.

File: app/networkapi/opportunities/templatetags/opportunities_adminsortable_tags_custom.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+@register.simple_tag(takes_context=True, name='render_object_rep')
+def render_object_rep(
+    context,
+    obj,
+    forloop
+):
+    context.update({'object': obj, 'forloop': forloop})
+    tmpl = template.loader.get_template(
+        'shared/adminsortable_objects_custom.html'
```
Comment (gideonthomas) [inline app/networkapi/opportunities/templatetags/opportunities_adminsortable_tags_custom.py]: see comment about duplicating templates

File: app/networkapi/opportunities/serializers.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+from rest_framework import serializers
+
+from networkapi.opportunities.models import Opportunity
+
+
+class OpportunitySerializer(serializers.ModelSerializer):
+    """
+    Serializes an Opportunity Model
+    """
+    link = serializers.SerializerMethodField('get_link_info')
```
Comment (gideonthomas) [inline app/networkapi/opportunities/serializers.py]: let's just use the default value, i.e. no need to specify an arg for this and just rename your method below to `get_link`

File: app/networkapi/templates/shared/adminsortable_objects_custom.html (hunk 1 above)
Comment (cadecairos) [inline app/networkapi/templates/shared/adminsortable_objects_custom.html]: I'll keep the class name generic, so we can share CSS, but we'll be able to customize the src string.

File: app/networkapi/static/people/js/admin.js (hunk 2 above)
Comment (cadecairos) [inline app/networkapi/static/people/js/admin.js]: woops.

File: app/networkapi/templates/shared/adminsortable_objects_custom.html (hunk 1 above)
Comment (gideonthomas) [inline app/networkapi/templates/shared/adminsortable_objects_custom.html]: I don't think we should be sharing anything in this file though. Like, some sortable models might not have images so an `img` tag will not be appropriate. This file itself is meant to serve as an html representation of a model/app so it is sorta counter-productive to have it be a general representation of all models/apps since we would need to make all sortable models have certain properties.

**Comment by gideonthomas** [comment]
@cadecairos let's just land these patches and disable them in production once they land.

File: app/networkapi/opportunities/models.py [hunk 6]
```diff
@@ -1,5 +1,42 @@
...
+                datetime(1970, 1, 1, tzinfo=timezone.utc)
+            ).total_seconds())
+        ),
+        ext=os.path.splitext(filename)[1]
+    )
+
+
+class Opportunity(SortableMixin):
+    name = models.CharField(max_length=300)
+    header = models.TextField(max_length=5000)
```
Comment (gideonthomas) [inline app/networkapi/opportunities/models.py]: let's remove this as it's no longer used in the frontend.

File: app/networkapi/opportunities/models.py [hunk 7]
```diff
@@ -1,5 +1,42 @@
...
+    description = models.TextField(max_length=5000)
+    link_text = models.CharField(max_length=300)
```
Comment (gideonthomas) [inline app/networkapi/opportunities/models.py]: let's call this `link_label` instead since that's what the frontend refers to it as.

File: app/networkapi/opportunities/models.py [hunk 8]
```diff
@@ -1,5 +1,42 @@
...
+    image = models.FileField(
+        max_length=2048,
+        upload_to=opportunity_image_path,
+    )
+    order = models.PositiveIntegerField(
+        default=0,
+        editable=False,
+        db_index=True,
+    )
+
```
Comment (gideonthomas) [inline app/networkapi/opportunities/models.py]: let's add `featured` here

File: app/networkapi/opportunities/serializers.py [hunk 9]
```diff
@@ -1,7 +1,25 @@
...
 class OpportunitySerializer(serializers.ModelSerializer):
-    pass
+    """
+    Serializes an Opportunity Model
+    """
+    link = serializers.SerializerMethodField()
+
+    def get_link(self, opportunity):
+        return {
+            'text': opportunity.link_text,
```
Comment (gideonthomas) [inline app/networkapi/opportunities/serializers.py]: let's change this to `label`

**Comment by gideonthomas** [comment]
Let's use https://github.com/mozilla/network/tree/master/source/json for schema from now on.

File: app/networkapi/opportunities/serializers.py [hunk 10]
```diff
@@ -1,7 +1,25 @@
...
+
+    def get_link(self, opportunity):
+        return {
+            'label': opportunity.link_label,
+            'url': opportunity.link_url,
+        }
+
+    class Meta:
+        model = Opportunity
+        fields = (
```
Comment (gideonthomas) [inline app/networkapi/opportunities/serializers.py]: this still needs to be updated.
